ansible_rulebook/action/control.py

```diff
@@ -0,0 +1,33 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import asyncio
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass(frozen=True)
+class Control:
```

Comment:
I feel this class needs a well described docstring.

Reply:
> @Alex-Izquierdo the **call** is the Callable feature from python that existed before
> 
> https://github.com/ansible/ansible-rulebook/blob/2830da5f1d25c075ed832341f510410296e2f6b7/ansible_rulebook/builtin.py#L889

I'm not following you, "Callable" type hint means that the object is callable, the code that you link is a dict containing regular functions, which of course are callable (everything that returns True with the function `callable(object)`

> These are internal actions we control how we execute it I don't think we plan on taking our actions and make it into a separate package. There is a tight coupling between actions and how we call it from the rule set runner.

I'm not saying that builtin actions will be in a separate package, I'm talking about what is the interface that we want for the actions now that we have the chance at the refactor, because in the future may be pluggable and I wonder if making the actions callable defining the __call__ internal method is the right way and it is aligned with ansible. In ansible for example we use explicit methods:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-a-module
https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#action-plugins

Curious to know the thoughts of @benthomasson